swagger.yaml

```diff
@@ -0,0 +1,262 @@
+swagger: "2.0"
+info:
+  description: "An API used to query information about datasets published by the ONS."
+  version: "1.0.0"
+  title: "ONS Dataset API"
+  license:
+    name: "Open Government Licence v3.0"
+    url: "http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
+basePath: "/v1"
+tags:
+- name: "Dataset"
+schemes:
+- "http"
+parameters:
+  id:
+    name: id
+    description: "Id that represents a dataset type"
+    in: path
+    type: string
+    required: true
+  edition:
+    name: edition
+    description: "An edition of a dataset"
+    in: path
+    type: string
+    required: true
+  version:
+    name: version
+    description: "A version of a dataset dataset"
+    in: path
+    type: string
+    required: true
+  dimension:
+    name: dimension
+    description: "A dimension from a dataset"
+    in: path
+    type: string
+    required: true
+  filter:
+    name: filter
+    description: "Filter a dataset using its dimensions"
+    in: body
+    schema:
+      $ref: '#/definitions/Filter'
+    required: true
+paths:
+  /datasets:
+    get:
+      tags:
+      - "Dataset"
+      summary: "Get all datasets"
+      description: "Returns a list of all datasets provided by the ONS that can be filtered using the job API"
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "A json list contain datasets which have been published"
+          schema:
+            $ref: '#/definitions/ArrayOfDatasets'
```

Comment:
I imagine this list will be paginated. Consider adding details of pagination to the spec. Not sure if this is something swagger supports?